cmd/mdatagen/internal/templates/component_test.go.tmpl

```diff
@@ -597,3 +597,21 @@ func generateLifecycleTestTraces() ptrace.Traces {
 }
 {{- end }}
 {{- end }}
+
+{{- if not .Tests.SkipLifecycle }}
+var _ component.Host = (*mdatagenNopHost)(nil)
+
+type mdatagenNopHost struct{}
```

Comment:
Some contrib components declare their own `nopHost` type, so I had to rename this type. Open to other names, I just went with an "obvious" name.